### ISSUE TITLE
Segfault on reinitialising connection - Reopened

### ISSUE THREAD

**Issue by AkhilNairAmey** [rstudio/sparklyr#1172]
Hi,

Posting a new issue as I can't seem to reopen the old one.

For some reason, #919 seems to have cropped up again.  I posted a `Dockerfile` reproducing the error which passed with the initial fix.  The same branch (`bugfix/reconnect-sigpipe`) which included a fix (and previously passed) now fails - with some additional spurious warnings, possibly from the `yaml` package.

The Dockerfile including the previous fix is below

```
FROM rocker/tidyverse:latest

RUN apt update && apt install r-cran-rjava -y

RUN apt-get install libc6-dbg gdb valgrind -y

RUN install2.r devtools

RUN R -e 'devtools::install_github("rstudio/sparklyr", ref = "bugfix/reconnect-sigpipe")'

RUN R -e 'sparklyr::spark_install("2.0.2")'

RUN printf "library(sparklyr)\n\nsc = spark_connect('local')\nspark_disconnect(sc)\nspark_connect('local')\n" > script.r
```

This now produces the output

```
* Using Spark: 2.0.2
Warning messages:
1: In yaml.load(readLines(con), error.label = error.label, ...) :
  R expressions in yaml.load will not be auto-evaluated by default in the near future
2: In yaml.load(readLines(con), error.label = error.label, ...) :
  R expressions in yaml.load will not be auto-evaluated by default in the near future
3: In yaml.load(readLines(con), error.label = error.label, ...) :
  R expressions in yaml.load will not be auto-evaluated by default in the near future
* Using Spark: 2.0.2

 *** caught segfault ***
address 0x5589327cc003, cause 'memory not mapped'
```

The same error occurs on the current master (`d49ea49`).

**Comment by crocker456** [comment]
I also experienced this same error.  really discouraging was working not too long ago.

**Comment by AkhilNairAmey** [comment]
@crocker456 apologies if you already know this, but you can put a `Sys.sleep(10)` before the reinit of the connection to avoid the error in the short term, so it's not all bad :)

**Comment by schlichtanders** [comment]
@crocker456 @AkhilNairAmey can you specify a previous sparklyr version which still worked?

**Comment by schlichtanders** [comment]
(the 10 seconds sleep does not work for me in some cases)

**Comment by AkhilNairAmey** [comment]
@schlichtanders, The `@bugfix/reconnect-sigpipe` ref previously contained the fix and worked.  This is unfortunately no longer the case, and it's unclear to me why (which is why I reopened the issue).
I can't help more than that (unless you provide a bit more info about your case) as the sleep is sufficient for me.

**Comment by javierluraschi** [comment]
@AkhilNairAmey fix available under `sparklyr 0.6.4-9904`

**Comment by schlichtanders** [comment]
thanks you so much!
